service.go

```
@@ -97,3 +97,24 @@ func (s *Service) Set(ctx context.Context, ttl time.Duration, key string, val in
 
 	return nil
 }
+
+// Exists returns true if the given key exists in redis.
+//
+// ctx should have a timeout
+//
+// key is the key that is being used to store the data
+func (s *Service) Exists(ctx context.Context, key string) (bool, error) {
+	const errMessage = "failed to get value from redis"
+
+	result, err := s.client.Exists(ctx, key).Uint64()
+	if err != nil {
+		return false, errors.Wrap(err, errMessage)
+	}
+
+	existsInRedis := result == 1
+	if !existsInRedis {
+		return false, errors.Wrap(ErrRedisKeyNotFound, errMessage)
```

I don't think this should throw any more errors, it's enough to know that the given key does not exist

---

if I were supposed to use this function and receive an error, I wouldn't want to check for this particular error every time just to ignore it afterwards

---

so just return the bool it is then